Add escape clause for slack usernames

haacked opened the pull request:
During a live online demo, I tried to use the `tweet` skill to tweet my own twitter username @haacked. Well, that happens to match my Slack username, so it replaced it with `<@U....>` (my slack mention).

Well that's no good!

Instead, we want to detect that and block it. We provide an escape hatch. Just use `@@haacked` instead.

haacked committed:
Add escape clause for slack usernames

During a live online demo, I tried to use the `tweet` skill to tweet my own twitter username @haacked. Well, that happens to match my Slack username, so it replaced it with `<@U....>` (my slack mention).

Well that's no good!

Instead, we want to detect that and block it. We provide an escape hatch. Just use `@@haacked` instead.

shana commented on tweet/main.py:
```diff
@@ -179,11 +179,17 @@ def tweet(cmd, param, words):
-        tweet_id = results.get("id")
-        bot.reply(":boom: just tweeted it! :point_right: https://twitter.com/{}/status/{}".format(screen_name, tweet_id))
+        # Replace @@haacked with @haacked. This gives us an escape hatch when a twitter username matches Slack.
+        tweet_text = re.sub(r'@@([a-zA-Z0-9_]+)', r'@\1', bot.arguments) 
```
Twitter usernames can only be between 1-15 characters long so you can probably do `@@([a-zA-Z0-9_]{1,15})` and be extra precise about it.

haacked replied:
Good call! I also need to do a look behind to make sure there’s a white space or beginning of string preceding `@@`.

haacked committed:
Improve twitter handle matching

Make sure we fix up:

`@@haacked`
`hello @@haacked`
`hello @@haacked, bye`

But ignore

`hello@@haacked`
